### DISCUSSION TITLE
Icons for new extensions like .park or .parkobj

### DISCUSSION THREAD

**Discussion by Richard-L** [OpenRCT2/OpenRCT2#17170] [Feature requests]
Would be nice to have icons in explorer distinguishing the different asset files. Currently they're a bit hard to tell apart:

![image](https://user-images.githubusercontent.com/4934209/167252268-a77e5522-5f3e-4820-bb54-d2a90a0bfabc.png)

I suppose the installer would need to register them across windows explorer, or other OSes respectively.

I tried searching a bit if someone had perhaps suggested this already but couldn't find. If you're interested I could probably try myself at generating some icons.

**Comment by tupaschoal** [comment]
I like the idea, maybe the park has some icon that resembles the map, and the same goes for a parkobj, but in "blueprinty"-style? 🤔

**Comment by Ryder17z** [comment]
One option is to recreate similar icons to what's present in rct2.exe
Shoddy example for .park files:
![image](https://user-images.githubusercontent.com/2000703/168622483-422141f0-76db-42eb-94ba-5fb837a38931.png)

**Comment by Richard-L** [comment]
Have OpenGraphics done any icons yet btw?

If I were to do some myself, what sizes would we want?

16x16 as the smallest I suppose, and then up to 64x64?

I suppose phones or tablets aren't enough of a thing yet for ORCT2 to warrant vector icons.

**Reply by Ryder17z** [reply]
There hasn't been anything made yet, apart from the openrct2 executable.

In the case of Android, there is nothing like for example MIME types on windows, meaning that you can't assign an icon to a particular file type.
Some file browsing apps might support changing the icon for a particular file type. From a quick serarch, it seems that Total Commander can do it, but that's not exactly reliable.

**Comment by 733737** [comment]
I offered to make icons for OpenGraphics, but the problem is that the game assets have to be extracted through a complex process which I don't understand, so getting the menu buttons into the game seems to be a nightmare since of the level of programmer knowledge needed and I couldn't find any tools to use that could extract or import game assets, so I was lost, and without reference gave up

**Reply by X123M3-256** [reply]
> I couldn't find any tools to use that could extract or import game assets, so I was lost, and without reference gave up

You can use OpenRCT2 itself to extract the graphics - just do `openrct2 sprite exportall /path/to/g1.dat /path/to/output/folder` and it will do it for you, there's no programming knowledge required. There are some third party tools that can extract the assets but many of them use a different color palette, I recommend using OpenRCT2 to do the extraction if you're creating sprites for OpenRCT2.

**Reply by 733737** [reply]
i think my main concern is how the data structure works in the game directory. Is there a guide somewhere on how to replace assets?

**Reply by IntelOrca** [reply]
@733737 icons for file extensions wouldn't be part of the game assets. For Linux, they probably need to be separate files distributed with the game and referenced by .desktop files. For Windows, icons are usually resources in the .exe or .dll file.

Just submit your icons as image files under the data directory or resources directory, and we can take it from there.

**Reply by 733737** [reply]
I didn't mean icons, i meant in game toolbar icons, although i can see how that should be a separate topic

**Reply by IntelOrca** [reply]
This discussion is about icons for file extensions in the OS file explorer.
